source/val/validate_function.cpp

```diff
@@ -24,6 +24,23 @@ namespace spvtools {
 namespace val {
 namespace {
 
+// When --relax-logical-pointer argument is given, we assume that types
+// of function parameter and argument are the same if they are both
+// OpTypePointer and those pointers point the same type regardless of
+// their storage class.
```

Comment:
This function should not care if the types are from function parameters or not.  The comments could be simpler: 

> Returns true if a| and |b| are instruction defining pointers that point to the same type.